### PR TITLE
Adding in the To-Do List (partial progress)

### PR DESCRIPTION
**Pull request by tamimfatahi**
Saving current progress for the to-do list implementation, a few errors/features need to be fixed/added:
  1. Highlight table header
  2. Actions Icons
  3. Actions themselves
  4. Switching pages
  5. Date Format
  6. Add checkmark to complete tasks by crossing-out/greying-out

### CODE REVIEW

**Comment by tamimfatahi** [comment]
Screenshot 2023-06-06 at 4 13 17 PM

**Comment by tamimfatahi** [comment]
Screenshot 2023-06-06 at 4 18 35 PM

**Comment by tamimfatahi** [comment]
Screenshot 2023-06-06 at 4 21 31 PM

**Review by seeddhee — COMMENTED** [review]
The code is readable and easy to navigate.
Make the difficulty linear for a better user experience.
Add an exit.

**Review by lemezze — COMMENTED** [review]
Great job with the MUI stuff! I'd suggest changing the formatting of the Date to make it more readable, and changing the Action buttons to make them symbols (i.e. red 'x' for cancel, green check for approval). I'd also like to see consistency in the Difficulty section text (i.e. choose either uppercase or lowercase). Other than that, looks good!

File: react-frontend/src/components/controls/Controls.js
Comment (dreassantos) [inline react-frontend/src/components/controls/Controls.js:1]: I like this. Is it so that any custom control can be used by calling Controlls?

**Comment by tamimfatahi** [comment]
> The code is readable and easy to navigate.
> Make the difficulty linear for a better user experience.
> Add an exit.

Will do in the next PR!

**Comment by tamimfatahi** [comment]
> Great job with the MUI stuff! I'd suggest changing the formatting of the Date to make it more readable, and changing the Action buttons to make them symbols (i.e. red 'x' for cancel, green check for approval). I'd also like to see consistency in the Difficulty section text (i.e. choose either uppercase or lowercase). Other than that, looks good!

I'll make sure to implement those. Thanks for the suggestions!

File: react-frontend/src/components/controls/Controls.js
Comment (tamimfatahi) [reply react-frontend/src/components/controls/Controls.js:1]: Yeah, all components are linked to controls for the sake of the organization. That way, you don't have to import all components individually wherever you want to utilize them.

**Pull request merged**